### ISSUE TITLE
Visual difference on Image styles between AMP and non-AMP pages

### ISSUE THREAD

**Issue by amedina** [ampproject/amp-wp#4419]
## Bug Description

Pair browsing of an AMP-first site in Transitional mode shows visual differences on image layouts.

## Expected Behaviour

There should be no visual disparities.

## Steps to reproduce

1. Activate the plugin in Transitional mode
2. Activate TwentyTwenty
3. Install Theme Unit Test data
4. Pair browse blog feed

## Additional context

Version: 1.5.0-RC1

<img width="1680" alt="Screen Shot 2020-03-22 at 9 55 06 PM" src="https://user-images.githubusercontent.com/506089/77285987-97bb9e80-6c8f-11ea-86a0-1f3a3f885d63.png">
<img width="1351" alt="Screen Shot 2020-03-22 at 10 04 53 PM" src="https://user-images.githubusercontent.com/506089/77286001-9be7bc00-6c8f-11ea-80a4-7f8d9c934982.png">
<img width="1344" alt="Screen Shot 2020-03-22 at 10 04 38 PM" src="https://user-images.githubusercontent.com/506089/77286004-9d18e900-6c8f-11ea-8089-6c1712a794ff.png">
<img width="1642" alt="Screen Shot 2020-03-22 at 9 56 22 PM" src="https://user-images.githubusercontent.com/506089/77286009-9e4a1600-6c8f-11ea-9ea8-4a4ff3332927.png">

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

**Comment by westonruter** [comment]
It's because the `amp-img` is getting `display:inline-block` whereas non-AMP img has `display:block`. In particular, it's because Twenty Twenty has [this CSS](https://github.com/WordPress/wordpress-develop/blob/630ba8979a319dc52da4ba1d3ded53cdf237be2a/src/wp-content/themes/twentytwenty/style.css#L621-L628):

```css
svg,
img,
embed,
object {
	display: block;
	height: auto;
	max-width: 100%;
}
```

However, an `amp-img` with `layout=intrinsic` gets:

```css
.i-amphtml-layout-intrinsic {
    display: inline-block;
    position: relative;
    max-width: 100%
}
```

I think what may be needed is adding a style rule to the core theme sanitizer specifically for Twenty Twenty that does this:

```css
amp-img[layout=intrinsic] {
    display: block;
}
```

**Comment by westonruter** [comment]
This is not a regression in 1.5-alpha. I tested in WordPress 5.0 with AMP plugin v1.4.x and it happens there as well.

**Comment by westonruter** [comment]
I have an initial fix in #4421 specific for Twenty Twenty. It works by prepending a style rule to `style[amp-custom]` which sets `amp-img` to have `display:block`, with a selector that is specific enough to be weightier than `.i-amphtml-layout-intrinsic`. 

There may be an opportunity to generalize this fix at the CSS parser level. For example, if coming across a selector that has just `img` and the rule includes `display:block`, then this could automatically result a style rule being added like:

```css
.amp-block-img { display:block }
```

While at the same time adding the `amp-block-img` class to every `amp-img`. This should be more robust/automatic, though I can imagine there would be more edge cases. Namely, images added via `amp-list` or `amp-script`.
